### PR TITLE
Add stopPropagation on keyEvents in md-menu

### PR DESCRIPTION
**Pull request by clshortfuse**
I also added a preventDefault on ESCAPE.

### CODE REVIEW

**Comment by ThomasBurleson** [comment]
@clshortfuse - what was the original issue that prompted this PR ?

**Comment by clshortfuse** [comment]
I have a table with rows that allows keyboard input for navigation. When the menu pop ups, the menu passes the up and down keys to its parent element (in this case the table). This isn't ideal. 

I can create a test if the explanation isn't clear.

**Comment by ThomasBurleson** [comment]
A unit test would be awesome. And how does `preventDefault()` help with ESC keys presses?

**Comment by clshortfuse** [comment]
It seems as though the issue is related to using a controller on the the body element, which I would have to do to support global commands. preventDefault() on would escape would stop the browser from doing it's regular action (stopped the page from loading).

Here's the unit test.
http://codepen.io/anon/pen/gpdxpq

Use up and down to navigate the table and press space bar to activate the menu. When you try to navigate the menu with up and down, those keys are passed back to the body. I'm also noticing that pressing spacebar on the menu selects the item, but also passes that through as well.

**Comment by ThomasBurleson** [comment]
@rschmukler - can you review. I will add unit tests if this is approved.

File: src/components/menu/menu-interim-element.js [hunk 1]
```diff
@@ -141,9 +141,9 @@ function MenuProvider($$interimElementProvider) {
         opts.menuContentEl.on('keydown', function(ev) {
           scope.$apply(function() {
             switch (ev.keyCode) {
-              case $mdConstant.KEY_CODE.ESCAPE: opts.mdMenuCtrl.close(); break;
-              case $mdConstant.KEY_CODE.UP_ARROW: focusMenuItem(ev, opts.menuContentEl, opts, -1); break;
-              case $mdConstant.KEY_CODE.DOWN_ARROW: focusMenuItem(ev, opts.menuContentEl, opts, 1); break;
+              case $mdConstant.KEY_CODE.ESCAPE: opts.mdMenuCtrl.close(); ev.stopPropagation(); ev.preventDefault(); break;
+              case $mdConstant.KEY_CODE.UP_ARROW: focusMenuItem(ev, opts.menuContentEl, opts, -1); ev.stopPropagation(); break;
+              case $mdConstant.KEY_CODE.DOWN_ARROW: focusMenuItem(ev, opts.menuContentEl, opts, 1); ev.stopPropagation(); break;
```
Comment (jelbourn) [inline src/components/menu/menu-interim-element.js]: Do up and down also not need preventDefault() ?

**Comment by clshortfuse** [comment]
I think I misunderstood preventDefault. I incorrectly assumed preventDefault was related to browser actions but this isn't true. You don't need preventDefault() if you do stopPropagation().

**Pull request closed without merging**